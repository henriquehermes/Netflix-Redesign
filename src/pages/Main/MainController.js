import React from 'react';

import MainComponent from './MainComponent';

import NETFLIX_ORIGINALS from '~/config/netflix_originals.json';
import TRENDING from '~/config/trending.json';
import CONTINUE_WATCHING from '~/config/continue_watching.json';

export default function MainController({ navigation }) {
  return (
    <MainComponent
      NETFLIX_ORIGINALS={NETFLIX_ORIGINALS}
      TRENDING={TRENDING}
      CONTINUE_WATCHING={CONTINUE_WATCHING}
      navigation={navigation}
    />
  );
}
