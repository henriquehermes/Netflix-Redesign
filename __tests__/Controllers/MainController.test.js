/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import MainController from '../../src/pages/Main/MainController';

const fakeLoginInApp = jest.fn();
const store = global.mockStore({
  movie: {
    detail: {},
  },
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MainController navigation={fakeLoginInApp} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
