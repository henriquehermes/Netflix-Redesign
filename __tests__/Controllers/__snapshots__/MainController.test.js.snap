// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#111111",
        "height": "100%",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": 70,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 0,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "marginRight": 10,
              },
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          漢
        </Text>
      </View>
    </View>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/img/Netflix_Logo.png",
        }
      }
      style={
        Array [
          Object {
            "height": 65,
            "width": 100,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "padding": 8,
              },
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 0,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                Object {
                  "marginRight": 10,
                },
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    overScrollMode="never"
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 18,
                "marginBottom": 15,
                "marginLeft": 25,
                "marginRight": 25,
                "marginTop": 0,
              },
            ]
          }
        >
          Netflix Originals
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#d7240f",
                  "fontSize": 12,
                  "marginBottom": 15,
                  "marginLeft": 25,
                  "marginRight": 25,
                  "marginTop": 0,
                },
              ]
            }
          >
            VIEW ALL
          </Text>
        </View>
      </View>
      <View
        height={210}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 210,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#D7240F"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 18,
              "marginBottom": 15,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            },
          ]
        }
      >
        Continue Watching
      </Text>
      <View
        height={191}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 191,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#D7240F"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 18,
              "marginBottom": 15,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            },
          ]
        }
      >
        Explore by Genres
      </Text>
      <View
        height={60}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 60,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#D7240F"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 18,
                "marginBottom": 15,
                "marginLeft": 25,
                "marginRight": 25,
                "marginTop": 0,
              },
            ]
          }
        >
          Trending
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#d7240f",
                  "fontSize": 12,
                  "marginBottom": 15,
                  "marginLeft": 25,
                  "marginRight": 25,
                  "marginTop": 0,
                },
              ]
            }
          >
            VIEW ALL
          </Text>
        </View>
      </View>
      <View
        height={108}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 108,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#D7240F"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 18,
              "marginBottom": 15,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            },
          ]
        }
      >
        Top 3 Movies Watched in Brazil 🇧🇷
      </Text>
      <View
        height={191}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 191,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#D7240F"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 18,
              "marginBottom": 15,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            },
          ]
        }
      >
        Top 3 Movies Watched in USA 🇺🇸
      </Text>
      <View
        height={191}
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 191,
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#D7240F"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
